### PR TITLE
Ensure that plan-preview tasks do not cause a panic on piped

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Ensure that plan-preview tasks do not cause a panic on piped
```

### CODE REVIEW

File: pkg/app/piped/planpreview/builder.go [hunk 1]
```diff
@@ -133,6 +144,12 @@ func (b *builder) Build(ctx context.Context, id string, cmd model.Command_BuildP
 
 	// Plan the trigger applications for more detailed feedback.
 	for _, app := range triggerApps {
+		b.logger.Info("will decide sync strategy for a application",
```
Comment (nakabonne) [inline pkg/app/piped/planpreview/builder.go:147]: nit
```suggestion
		b.logger.Info("will decide sync strategy for an application",
```

File: pkg/app/piped/planpreview/builder.go [hunk 2]
```diff
@@ -95,7 +95,18 @@ func newBuilder(
 	}
 }
 
-func (b *builder) Build(ctx context.Context, id string, cmd model.Command_BuildPlanPreview) ([]*model.ApplicationPlanPreviewResult, error) {
+func (b *builder) Build(ctx context.Context, id string, cmd model.Command_BuildPlanPreview) (results []*model.ApplicationPlanPreviewResult, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("an unexpected panic occurred (%w)", r)
```
Comment (nakabonne) [inline pkg/app/piped/planpreview/builder.go:101]: nit: If `(%w)` is more idiomatically, fine to be as is.
```suggestion
			err = fmt.Errorf("an unexpected panic occurred: %w", r)
```

File: pkg/app/piped/planpreview/builder.go (hunk 1 above)
Comment (nghialv) [reply pkg/app/piped/planpreview/builder.go]: 👍

File: pkg/app/piped/planpreview/builder.go (hunk 2 above)
Comment (nghialv) [reply pkg/app/piped/planpreview/builder.go:101]: Let me keep it as it is because we are applying the same way at other places too.

Comment (nakabonne) [reply pkg/app/piped/planpreview/builder.go:101]: in that case that's fine

**Comment by nakabonne** [comment]
Thank you
/approve

**Pull request merged**